src/tools/synth_node.rs

```diff
@@ -85,6 +86,23 @@ impl SyntheticNode {
         }
     }
 
+    /// Reads a message from the inbound (internal) queue of the node. If there is no message
+    /// by the given time there is an error returned indicating if timeout occured.
+    pub async fn recv_message_timeout(
```

Comment:
I really like the approach [zcash/algorand](https://github.com/runziggurat/zcash/blob/main/src/tools/synthetic_node.rs#L282) chose here for the same function. Could we reuse it? 🤔

Reply:
Reused it directly.